cmd/ncproxy/ncproxy.go

```diff
@@ -630,18 +503,22 @@ func (s *grpcService) GetNetworks(ctx context.Context, req *ncproxygrpc.GetNetwo
 	defer span.End()
 	defer func() { oc.SetSpanStatus(span, err) }()
 
+	networks := []*ncproxygrpc.GetNetworkResponse{}
+
 	rawNetworks, err := hcn.ListNetworks()
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get HNS networks")
 	}
 
-	networks := make([]*ncproxygrpc.GetNetworkResponse, len(rawNetworks))
-	for i, network := range rawNetworks {
-		resp := &ncproxygrpc.GetNetworkResponse{
-			ID:   network.Id,
+	for _, network := range rawNetworks {
+		networkReq := &ncproxygrpc.GetNetworkRequest{
 			Name: network.Name,
 		}
-		networks[i] = resp
+		n, err := s.GetNetwork(ctx, networkReq)
```

Comment:
would this `err :=` assignment overshadow the one used to set the span status?

Reply:
No because if this error is not nil, it will be returned and then that error is the one used to set the span status. Does that make sense? 

The named returned value for err will be set to the error that is returned from this function.

Reply:
yup 👍